packages/web-client/app/components/card-space/create-space-workflow/index.ts

```diff
@@ -328,7 +328,9 @@ export default class CreateSpaceWorkflowComponent extends RestorableWorkflowComp
       profilePhoto: this.workflow.session.getValue('profileImageUrl'),
       coverPhoto: this.workflow.session.getValue('coverPhotoUrl'),
       name: this.workflow.session.getValue('displayName'),
-      host: this.workflow.session.getValue('url'),
+      host: this.workflow.session.getValue('merchantId')
```

Comment:
I tried using `useResource` here (as an alternative to the fetching operations in the previous commits) like this:

```
get merchantData() {
    let did = this.workflow.session.getValue('merchantInfoDID');
    
    if (did) {
      return useResource(this, MerchantInfo, () => ({
        infoDID: did,
      }));
    } else {
      return null;
    }
  }
  
  get currentCardSpaceDetails() {
    return {
      ...
      host: this.merchantData ? `${this.merchantData.id}.card.space` : null,
      ...
    };
  }
  ```
  
  ... but this ends up in in an endless loop (the merchantData is called endlessly)  😵‍💫 Any ideas why that would happen?

Reply:
I think the problem is that `useResource` is meant to be like a service injection or other property, not something used within a function. I tried this and it worked:

![image](https://user-images.githubusercontent.com/43280/151881890-d06074a7-816e-49be-b726-7c8b5bbfaf0f.png)

As an aside, `ember-resources` now has a [`@use` decorator](https://github.com/NullVoxPopuli/ember-resources#use) that we should probably start `use`ing 😆

Reply:
Thanks! I used the `useResource` which made this PR much simpler! 🎉